queue_/queue_.py

```
@@ -13,40 +14,58 @@ class Queue_:
     Queue Data Structure
     """
 
-    def __init__(self, data: Iterable = ()):
-        pass
+    def __init__(self, data: Iterable = (), max_size: int = 0):
+        self.data = list(data)
+        self.max_size = max_size
 
     def put(self, element):
         """
         Add the element ‘element’ at the end of queue_
         :param element: element to add to queue_
         """
-        pass
+        if self.full():
+            raise Full
+
+        self.data.append(element)
 
     def get(self):
         """
         Remove and return an item from queue_
         """
-        pass
+        return self.data.pop(0)
 
     def empty(self) -> bool:
         """
         Return whether queue_ is empty or not
         :return: True if queue_ does not contain any elements.
                  False if the queue_ contains elements
         """
-        pass
+        return not self.data
 
     def size(self) -> int:
         """
         Return the number of elements in queue_
         :return: Number of elements in queue_
         """
-        pass
+        return len(self.data)
 
     def top(self):
         """
         Return the element on the top of queue_
         :return: the element that is on the top of queue_
         """
-        pass
+        return self.data[0]
+
+    def full(self):
+        """
+        Return whether queue_ is full or not
+        :return: True if queue_ is full.
+                 False if the queue_ is not full
+        """
+        if not self.max_size:
+            print('the queue size is infinite')
```

you can return it

---

return a string? i thought it was supposed to only return True or False

---

fair enough
but printing is not the best option
you can raise an error, for example